### PR TITLE
Add PolyNet

### PR DESCRIPTION
**Pull request by creafz**
PolyNet model from [PolyNet: A Pursuit of Structural Diversity in Very Deep Networks](https://arxiv.org/abs/1611.05725).

Weights are ported from the [Caffe implementation](https://github.com/CUHK-MMLAB/polynet). Accuracy values for the original model were taken from the paper (Table 1).

PyTorch weights: https://drive.google.com/file/d/1gzAvavIaxRG38cOkuWb-NxjBHJ6YGqE4/view

A note on the evaluation: as far as I understand to evaluate the performance of the model on the ImageNet evaluation set the authors resized images to 378x378 without maintaining the aspect ratio and then took a center 331x331 crop. To mimic this behavior I have to modify `imagenet_eval.py` and add the `--do-not-preserve-aspect-ratio` argument, but I am not sure maybe there are better ways of handling this, what do you think?

For the modified command (`python examples/imagenet_eval.py --data /path/to/imagenet -a polynet -b 64 -e --do-not-preserve-aspect-ratio`) I got `Acc@1 81.002 Acc@5 95.624`. When the standard evaluation that preserves the aspect ratio is used resulting accuracy values for the model are slightly worse:  `Acc@1 80.858 Acc@5 95.560`

### CODE REVIEW

**Comment by Cadene** [comment]
To handle the special preprocessing, we could create a new key to the pretrained_settings dictionary. But I am ok with the `--do-not-preserve-aspect-ratio` flag for now. The note is there. If anyone needs it we can add it in a near future.

Thanks!

**Comment by creafz** [comment]
@Cadene PolyNet model weights have the same PyTorch 0.3 compatibility issue as PNasNet in #65, so here is an updated version of weights that should be compatible both with PyTorch 0.3.1 and 0.4, could you test it in your environment?

https://drive.google.com/file/d/1Zj_RzCnIFqZgYLieE3sPdRRE8b3D-89S/view

**Comment by Cadene** [comment]
@creafz Done! https://github.com/Cadene/pretrained-models.pytorch/commit/3acaa88d8145db265b5e1dec03e91eb751b82c74
